### PR TITLE
subsys: logging: defines clean up

### LINKED ISSUE #14536: out of bounds access in log_backend_rtt
When building with clang we get these warnings, which seem like a real possible issue.  Seems like we might need some ifdef protection.

This was seen building:

-p nrf52840_pca10056 -T samples/basic/fade_led

```
/home/galak/git/zephyr/subsys/logging/log_backend_rtt.c:116:4: error: array index -2 is before the beginning of the array [-Werror,-Warray-bounds]
                        line_buf[DROP_MSG_LEN - 2] = ' ';
                        ^        ~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/subsys/logging/log_backend_rtt.c:49:1: note: array 'line_buf' declared here
static u8_t line_buf[MESSAGE_SIZE + DROP_MSG_LEN];
^
/home/galak/git/zephyr/subsys/logging/log_backend_rtt.c:117:4: error: array index -3 is before the beginning of the array [-Werror,-Warray-bounds]
                        line_buf[DROP_MSG_LEN - 3] = (u8_t) ('0' + cnt);
                        ^        ~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/subsys/logging/log_backend_rtt.c:49:1: note: array 'line_buf' declared here
static u8_t line_buf[MESSAGE_SIZE + DROP_MSG_LEN];
^
/home/galak/git/zephyr/subsys/logging/log_backend_rtt.c:118:4: error: array index -4 is before the beginning of the array [-Werror,-Warray-bounds]
                        line_buf[DROP_MSG_LEN - 4] = ' ';
                        ^        ~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/subsys/logging/log_backend_rtt.c:49:1: note: array 'line_buf' declared here
static u8_t line_buf[MESSAGE_SIZE + DROP_MSG_LEN];
^
/home/galak/git/zephyr/subsys/logging/log_backend_rtt.c:120:4: error: array index -2 is before the beginning of the array [-Werror,-Warray-bounds]
                        line_buf[DROP_MSG_LEN - 2] = (u8_t) ('0' + cnt % 10);
                        ^        ~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/subsys/logging/log_backend_rtt.c:49:1: note: array 'line_buf' declared here
static u8_t line_buf[MESSAGE_SIZE + DROP_MSG_LEN];
^
/home/galak/git/zephyr/subsys/logging/log_backend_rtt.c:121:4: error: array index -3 is before the beginning of the array [-Werror,-Warray-bounds]
                        line_buf[DROP_MSG_LEN - 3] = (u8_t) ('0' + cnt / 10);
                        ^        ~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/subsys/logging/log_backend_rtt.c:49:1: note: array 'line_buf' declared here
static u8_t line_buf[MESSAGE_SIZE + DROP_MSG_LEN];
^
/home/galak/git/zephyr/subsys/logging/log_backend_rtt.c:122:4: error: array index -4 is before the beginning of the array [-Werror,-Warray-bounds]
                        line_buf[DROP_MSG_LEN - 4] = '>';
                        ^        ~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/subsys/logging/log_backend_rtt.c:49:1: note: array 'line_buf' declared here
static u8_t line_buf[MESSAGE_SIZE + DROP_MSG_LEN];
^
6 errors generated.
make[3]: *** [zephyr/CMakeFiles/zephyr.dir/build.make:414: zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_backend_rtt.c.obj] Error 1
```

### PR DESCRIPTION
**Pull request by nordic-krch**
RTT backend supports two modes blocking and drop. Apparently,
defines used lead to warning while clang compilation. Define
that caused warning has been changed together with clean up
which removed #ifdefs for definitions.

Fixes #14536 

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

### CODE REVIEW

**Comment by nordic-krch** [comment]
@galak could you try it with clang?

File: subsys/logging/log_backend_rtt.c [hunk 1]
```diff
@@ -10,39 +10,49 @@
...
+#endif
+
+#ifndef CONFIG_LOG_BACKEND_RTT_RETRY_CNT
+/* Big enough to detect host presence */
+#define CONFIG_LOG_BACKEND_RTT_RETRY_CNT 10
+#endif
+
+#define DROP_MAX 99
 
 #define DROP_MSG "\nmessages dropped:    \r"
```
Comment (jarz-nordic) [inline subsys/logging/log_backend_rtt.c]: I know that it is not part of this change, but do you know what is a purpose of `\r` at the end?

**Comment by galak** [comment]
@nordic-krch any plans to update?

File: subsys/logging/log_backend_rtt.c (hunk 1 above)
Comment (nordic-krch) [reply subsys/logging/log_backend_rtt.c]: nope,

**Comment by nordic-krch** [comment]
@galak did you tried it with clang. Is the warning fixed?

**Review by galak — APPROVED** [review]

**Pull request merged**